### PR TITLE
Change `sum` to `_sum` in `kron` benchmarks 

### PR DESCRIPTION
**Pull request by yebai**

### CODE REVIEW

**Comment by willtebbutt** [comment]
Why?

**Comment by willtebbutt** [comment]
Ah okay. `kron` and friends suffer the same problems as `sum`, so you might see a small improvement in the benchmarks, but I doubt that it will be very substantial. Interested to see though.

File: bench/run_benchmarks.jl [hunk 1]
```diff
@@ -60,7 +60,7 @@ end
 
 # Zygote has rules for both sum and kron, so it's interesting to compare this against the
 # other frameworks because they don't have rules for kron, and maybe for not for sum.
-_kron_sum(x::AbstractMatrix{<:Real}, y::AbstractMatrix{<:Real}) = sum(kron(x, y))
+_kron_sum(x::AbstractMatrix{<:Real}, y::AbstractMatrix{<:Real}) = _sum(kron(x, y))
```
Comment (yebai) [inline bench/run_benchmarks.jl]:
```suggestion
_kron_sum(x::AbstractMatrix{<:Real}, y::AbstractMatrix{<:Real}) = _sum(identity, kron(x, y))
```

**Comment by willtebbutt** [comment]
It looks like Zygote _really_ doesn't like `_sum`. You might want to consider disabling Zygote for this run. You can do so by adding this method to the benchmark suite:
```julia
should_run_benchmark(::Val{:zygote}, ::Any, x) = false
```

**Pull request closed without merging**